### ISSUE TITLE
Error code from spacefm:17519

### ISSUE THREAD

**Issue by RaveYoda** [IgnorantGuru/spacefm#486]
OS: Ubuntu 14.04
Window Manager: PekWM
SpaceFM version: 0.9.9+
Theme Type: GTK-2.0

Spacefm --desktop seems to crash if I close, open, close, then re-open just the spacefm manager.
---> Terminal output recieved when my primary theme is in place:
(spacefm:17519): Gtk-WARNING **: Unable to locate theme engine in module_path: "ubuntulooks",
(spacefm:17519): Gtk-WARNING **: Theme directory  of theme Azenis Icons has no size field
(spacefm:17519): Gtk-CRITICAL **: IA__gtk_style_detach: assertion 'style->attach_count > 0' failed
(spacefm:17519): Gtk-CRITICAL **: IA__gtk_style_detach: assertion 'style->attach_count > 0' failed
Could not open input file: /home/abaddon/Dropbox/Movies-Music/Carl Orff-O Fortuna~Carmina Burana.webm
(spacefm:17519): GLib-GObject-CRITICAL **: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
(spacefm:17519): GLib-GObject-CRITICAL **: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
(spacefm:17519): GLib-GObject-CRITICAL **: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
(spacefm:17519): GLib-GObject-CRITICAL **: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
(spacefm:17519): GLib-GObject-CRITICAL **: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
(spacefm:17519): GLib-GObject-CRITICAL **: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
(spacefm:17519): GLib-GObject-CRITICAL **: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
(spacefm:17519): Gtk-CRITICAL **: _gtk_style_peek_property_value: assertion 'GTK_IS_STYLE (style)' failed
Segmentation fault (core dumped)

---> Terminal output recieved when my primary theme is in NOT used:
(spacefm:17915): GLib-GObject-CRITICAL **: g_object_unref: assertion 'object->ref_count > 0' failed
(spacefm:17915): GLib-GObject-CRITICAL **: g_object_unref: assertion 'object->ref_count > 0' failed
(spacefm:17915): GLib-GObject-CRITICAL **: g_object_ref: assertion 'G_IS_OBJECT (object)' failed

After changing my primary theme to several other ones it would take about 6 or so reopins befor crashing the desktop Icons. So...it seemed to inprove somewhat. I'm just not certain if spacefm is having issues or it's my selection of themes.

**Comment by IgnorantGuru** [comment]
This almost certainly looks like a theme problem.  Usually this is more characteristic of breakage in GTK3, when the theme's version and GTK's version mismatch.  You said GTK2 (you can confirm this with spacefm --version).  So maybe a corrupted or mismatched theme.

For GTK2, there are some suggestions for how to test themes in the [user's manual](http://ignorantguru.github.io/spacefm/spacefm-manual-en.html#invocation-gtkthemes).

In the README there are instructions for getting a backtrace on the crash (see BUILD DEBUG section), although in this case it looks like the theme.

**Comment by RaveYoda** [comment]
OK, I'll check it out further. Just wanted your input on it. =] Thanks!

**Comment by RaveYoda** [comment]
I changed over to using GTK3 SpaceFM and the desktop icons no longer crash after multiple open/closeings of SpaceFM. I tried several GTK2 themes prior to see if it would solve my issue and nothing worked definitavly. I'd still get errors like bellow :
(spacefm:17915): GLib-GObject-CRITICAL *: g_object_unref: assertion 'object->ref_count > 0' failed
(spacefm:17915): GLib-GObject-CRITICAL *: g_object_unref: assertion 'object->ref_count > 0' failed
(spacefm:17915): GLib-GObject-CRITICAL **: g_object_ref: assertion 'G_IS_OBJECT (object)' failed
So I'll just stick with the GTK3 version. XD

**Comment by IgnorantGuru** [comment]
On GTK2, it's best to test with either the Adwaita theme or with no theme at all.  I can't reproduce those messages

**Comment by RaveYoda** [comment]
Ahhhh, yeah, I wasn't using that theme at all during testing. XD <Shrugs> It's all good man. I just wanted to update you on what was going on because I saw the label for this report as "Need Info". I closed this report for you. Thanks for double checking your end buddy!! I really am glad you've resumed work on SpaceFM as this is my favorite file manager!! I've yet to find anything that begins challenging this program. =]
